src/strategies/gains-vault-staked-balance/index.ts

```diff
@@ -0,0 +1,34 @@
+import { BigNumberish } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+import { Multicaller } from '../../utils';
+
+export const author = 'onecri';
+export const version = '0.0.1';
+
+const abi = [
+  'function userDeposit(address user) external view returns (uint256, uint256)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const multi = new Multicaller(network, provider, abi, { blockTag });
+  addresses.forEach((address) =>
+    multi.call(address, options.vault, 'userDeposit', [address])
+  );
+  const result: Record<string, BigNumberish> = await multi.execute();
+
+  return Object.fromEntries(
+    Object.entries(result).map(([address, res]) => [
+      address,
+      parseFloat(formatUnits(res[0], options.decimals))
+    ])
+  );
+}
```

Comment:
@onecri I think the `contract-call` strategy should work in this case, no need for a separate strategy :) Can try? let me know

Reply:
HI @ChaituVR  i want to get our own separate strategy. Is it impossible?

Reply:
@onecri Is there any reason why?

Reply:
We wanted to reduce duplicate strategies as much as possible. so we added a checklist here https://github.com/snapshot-labs/snapshot-strategies#code

Reply:
@ChaituVR there is no special reason, but we want to have our own.

Reply:
I understand that you want to have your own strategy easily identifiable by the space's name, however, we highly value the core development principles of keeping the code DRY. We are aiming to minimize the repetitive code and at the same time provide the best experience to our users (lack of confusion regarding the difference between two strategies with the same functionality, reusability of available tools, recognition of popular strategies among voters, etc). 

If the strategy requires custom implementation, then it makes total sense to create a new one however in your case we would appreciate it if you could use what has already been implemented and tested by multiple spaces.

Reply:
Ok, got it

Reply:
![image](https://user-images.githubusercontent.com/52106820/217033271-3916a6bc-601e-4718-a4e1-1537e6b64baa.png)

Reply:
I am trying with `contract-call`, but i got this error. I believe this error is because the length of the return value of `userDeposit` function is 2 ([uint256, uint256]). Is there any way to get only 1st value from returned array? @ChaituVR

Reply:
Ok, thank you